### PR TITLE
Correctly identify field for VALUE_OUT_OF_RANGE errors

### LINKED ISSUE #27565: VALUE_OUT_OF_RANGE error misattributes to m2o uuid field when a different numeric column overflows
## Describe the Bug

When a POST payload contains a numeric value that exceeds a column's precision/scale, Directus correctly detects the overflow and throws `VALUE_OUT_OF_RANGE` — but the error's `field` extension points to an unrelated m2o uuid field in the same payload, not the actual numeric column that overflowed.

This makes the error message actively misleading. The reported field's value is a valid uuid and its column type is uuid; the actual fault is elsewhere in the payload.

## To Reproduce

Schema (typical of what the Directus UI produces when you pick "Decimal" with default precision):

```sql
CREATE TABLE inventory (
  id uuid PRIMARY KEY,
  stock_number varchar(255),
  retail_price numeric(10, 5),
  body_type_id uuid REFERENCES body_types(id)
);
```

`body_type_id` is configured in Directus admin as a many-to-one to a `body_types` collection.

POST to `/items/inventory`:

```json
{
  "stock_number": "TEST01",
  "retail_price": 224266,
  "body_type_id": "f89834b4-682f-41f1-b48f-2375c5cdd618"
}
```

`retail_price` overflows `numeric(10, 5)` (max value 99999.99999).

## Expected Behavior

Error reported against `retail_price`:

```json
{
  "errors": [{
    "message": "Numeric value \"224266\" for field \"retail_price\" in collection \"inventory\" is out of range.",
    "extensions": {
      "collection": "inventory",
      "field": "retail_price",
      "value": "224266",
      "code": "VALUE_OUT_OF_RANGE"
    }
  }]
}
```

## Actual Behavior

Error reported against `body_type_id` (the m2o uuid field):

```json
{
  "errors": [{
    "message": "Numeric value \"f89834b4-682f-41f1-b48f-2375c5cdd618\" for field \"body_type_id\" in collection \"inventory\" is out of range.",
    "extensions": {
      "collection": "inventory",
      "field": "body_type_id",
      "value": "f89834b4-682f-41f1-b48f-2375c5cdd618",
      "code": "VALUE_OUT_OF_RANGE"
    }
  }]
}
```

Additional observations:

- Removing `body_type_id` from the payload moves the error to whichever other m2o uuid field is present (e.g. `brand_id`) with the same misleading framing.
- Removing `retail_price` from the payload makes the error disappear entirely — confirming `retail_price` is the real source.
- The `body_type_id` column's `data_type` is correctly `uuid` in both Postgres and Directus's own `/fields/inventory/body_type_id` metadata response, with `numeric_precision: null` and `numeric_scale: null`. Nothing in the field's own schema explains the "numeric" framing.

## Impact

The misattribution caused several hours of debugging — the reported field's schema is internally consistent and obviously not numeric, so the natural assumption is that the validator has stale schema state for that field. Repeatedly verifying the schema while the validator kept reporting `body_type_id` as out-of-range was deeply confusing.

Worth noting as adjacent context: the Directus UI's default precision for the "Decimal" field type is `(10, 5)`, which is a sharp edge for currency fields where 5 fractional digits is unusual and 5 integer digits is restrictive (max value 99,999.99999). This makes the underlying validation correct but the misattribution more likely to be hit in the wild.

## Environment

- Directus: 11.17.0
- Database: PostgreSQL 16 (`postgres:16-alpine`)
- Self-hosted Docker Compose deployment
- Behind Cloudflare with Caddy reverse proxy (not relevant to repro, just noting the setup)

### Directus Version

11.17.0

### Hosting Strategy

Self-Hosted (Docker Image)

### Database

PostgreSQL 16

### PR DESCRIPTION
**Pull request by sourav-18**
## Summary

PostgreSQL does not specify which column caused a `VALUE_OUT_OF_RANGE` error. The previous implementation relied on parsing the first identifier from the database error message, which could incorrectly attribute the error to another field (such as an m2o UUID field).

While investigating this issue, I also found that `numeric_precision` and `numeric_scale` existed in the schema types but were not populated by the schema package. This PR populates those values and uses them to identify the correct field when a numeric value exceeds its allowed range.

fix #27565

### CODE REVIEW

File: api/src/database/errors/dialects/postgres.ts [hunk 1]
```diff
@@ -58,12 +62,33 @@ export function extractError(error: PostgresError, data: Partial<Item>): Postgre
...
+		if (collection && schema && schema.collections[collection]) {
+			for (const [key, value] of Object.entries(data)) {
+				const fieldInfo = schema.collections[collection].fields[key];
+
+				if (fieldInfo?.dbType !== 'numeric' || typeof value !== 'string') continue;
+
+				const [integerPart = '', decimalPart = ''] = value.split('.');
+
+				if (
+					(fieldInfo.precision && integerPart.length > (fieldInfo.precision ?? 0) - (fieldInfo.scale ?? 0)) ||
```
Comment (sviat-barbutsa) [inline api/src/database/errors/dialects/postgres.ts:78]: I think this needs one more edge-case test around signed numeric values because the integer digit check currently uses `integerPart.length`, but `integerPart` still includes a leading `-` or `+`. For example, with a `numeric(5, 2)` field, `-123.45` is valid in Postgres because the sign is not part of the precision, but this check sees `integerPart === "-123"` and counts it as 4 integer digits.

That can misidentify the offending field if the payload contains a valid negative numeric field before another field that actually caused the `VALUE_OUT_OF_RANGE` error. Could you, please, normalize the string before counting integer digits, at least stripping a leading sign, and add a regression test with a valid negative numeric value plus a second overflowing numeric field?

Comment (sourav-18) [reply api/src/database/errors/dialects/postgres.ts:78]: Thanks @sviat-barbutsa , that's a valid point.

However, in this implementation the precision/scale check only runs after PostgreSQL returns a `VALUE_OUT_OF_RANGE` error. Since `-123.45` is a valid value for `numeric(5,2)`, PostgreSQL accepts it successfully and this code path is never reached.

**Comment by ComfortablyCoding** [comment]
Thanks for the PR, closing in favor of #27690
